server/index.ts

```diff
@@ -47,125 +56,255 @@ app.get('/auth/login', (_req, res) => {
     scope: scope.join(' '),
     redirect_uri: spotifyRedirectUri,
     state: state,
-  } as URLProps);
+  } as LoginUrlProps);
 
   res.redirect(
     `https://accounts.spotify.com/authorize/?${authQueryParameters.toString()}`,
   );
 });
 
-// callback endpoint
-app.get('/auth/callback', (req, res) => {
+// callback endpoint -- set token server globals and redirect to homepage
+app.get('/auth/callback', async (req, res) => {
   const { code } = req.query;
+  const formData = new URLSearchParams({
+    code,
+    redirect_uri: spotifyRedirectUri,
+    grant_type: 'authorization_code',
+  } as CbUrlProps).toString();
 
-  const authOptions = {
-    url: 'https://accounts.spotify.com/api/token',
-    form: {
-      code: code,
-      redirect_uri: spotifyRedirectUri,
-      grant_type: 'authorization_code',
-    },
-    headers: {
-      Authorization: `Basic ${Buffer.from(
-        `${spotifyClientId}:${spotifyClientSecret}`,
-      ).toString('base64')}`,
-      'Content-Type': 'application/x-www-form-urlencoded',
-    },
-    json: true,
-  };
-
-  request.post(authOptions, function (error, response, body) {
-    if (!error && response.statusCode === 200) {
-      accessToken = body.access_token;
-      res.redirect('http://localhost:3000');
-    } else {
-      console.log(
-        error,
-        response.body.error,
-        response.body.error.error_description,
-      );
-      res.end(response.body);
-    }
-  });
+  try {
+    const response = await axios({
+      method: 'post',
+      url: 'https://accounts.spotify.com/api/token',
+      data: formData,
+      headers: {
+        Authorization: `Basic ${Buffer.from(
+          `${spotifyClientId}:${spotifyClientSecret}`,
+        ).toString('base64')}`,
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+    });
+    accessToken = response.data.access_token;
+    expiresIn = response.data.expires_in;
+    refreshToken = response.data.refresh_token;
```

Comment:
Destructure me!!!
```suggestion
  try {
    const { data: { access_token, expires_in, refresh_token }} = await axios({
      method: 'post',
      url: 'https://accounts.spotify.com/api/token',
      data: formData,
      headers: {
        Authorization: `Basic ${Buffer.from(
          `${spotifyClientId}:${spotifyClientSecret}`,
        ).toString('base64')}`,
        'Content-Type': 'application/x-www-form-urlencoded',
      },
    });
    accessToken = access_token;
    expiresIn = expires_in;
    refreshToken = refresh_token;
```

Reply:
eslint had me take it a step further and convert destructured naming convention to an 'acceptable' one (camelCase, pascal, or upper).